transitions/index.js

```diff
@@ -103,6 +117,83 @@ const processChange = (change, callback) => {
     });
 };
 
+const deleteInfoDoc = (change, callback) => {
+  db.medic.get(`${change.id}-info`, (err, doc) => {
+    if (err) {
+      if (err.statusCode === 404) {
+        // don't worry about deleting a non-existant doc
+        return callback();
+      }
+      return callback(err);
+    }
+    doc._deleted = true;
+    db.medic.insert(doc, callback);
+  });
+};
+
+const getAdminNames = callback => {
+  const options = { key: [ 'user-settings' ], include_docs: true };
+  db.medic.view('medic-client', 'doc_by_type', options, (err, results) => {
+    if (err) {
+      return callback(err);
+    }
+    const admins = results.rows
+      .filter(row => {
+        return row.doc.roles &&
+               (row.doc.roles.includes('_admin') ||
+               row.doc.roles.includes('national_admin'));
+      })
+      .map(row => {
+        const parts = row.id.split(':');
+        return parts.length > 1 && parts[1];
+      })
+      .filter(name => !!name);
+    callback(null, admins);
+  });
+};
+
+const deleteReadDocs = (change, callback) => {
+
+  // we don't know if the deleted doc was a report or a message so
+  // attempt to delete both
+  const possibleReadDocIds = [ 'report', 'message' ]
+    .map(type => `read:${type}:${change.id}`);
+
+  getAdminNames((err, admins) => {
+    if (err) {
+      return callback(err);
+    }
+
+    async.each(
+      admins,
+      (admin, callback) => {
+        const metaDb = db.use(`medic-user-${admin}-meta`);
+        metaDb.info(err => {
+          if (err) {
+            if (err.statusCode === 404) {
+              // no meta data db exists for this user - ignore
+              return callback();
+            }
+            return callback(err);
+          }
+          metaDb.fetch({ keys: possibleReadDocIds }, (err, results) => {
+            if (err) {
+              return callback(err);
+            }
+            const row = results.rows.find(row => !!row.doc);
```

Comment:
Not sure why you'd need to do this line. When would you have a row that didn't also contain a doc?

Oh, is this because it returns 404 rows for each key you fetch?

If so, maybe a comment would make this clearer to the reader?

Reply:
Done.